### PR TITLE
feat(createWebhook): webhook name validation

### PR DESCRIPTION
**Pull request by vitorlops**
**Please describe the changes this PR makes and why it should be merged:**
This PR adds a webhook name validation as per
- https://github.com/discord/discord-api-docs/pull/4625

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes

### CODE REVIEW

File: packages/discord.js/src/managers/GuildChannelManager.js [hunk 1]
```diff
@@ -191,9 +191,17 @@ class GuildChannelManager extends CachedManager {
...
     if (!id) throw new TypeError('INVALID_TYPE', 'channel', 'GuildChannelResolvable');
+
     if (typeof avatar === 'string' && !avatar.startsWith('data:')) {
       avatar = await DataResolver.resolveImage(avatar);
     }
+
+    const invalidNames = ['everyone', 'here'];
+    const invalidSubstrings = ['@', '#', ':', '```', 'discord', 'clyde'];
+    if (invalidNames.includes(name) || invalidSubstrings.some(c => name.includes(c))) {
+      throw new TypeError('INVALID_WEBHOOK_NAME');
```
Comment (vladfrangu) [inline packages/discord.js/src/managers/GuildChannelManager.js:202]: Is this error message registered? If not, it should be

**Review by vladfrangu — CHANGES_REQUESTED** [review]
I don't really know if we want to do this validation (as we don't do..a lot of validation for items that might change at the API level + confusables being usable for bypassing), but this is the only thing I noticed that needs to be fixed if we do go forward with this

File: packages/discord.js/src/managers/GuildChannelManager.js (hunk 1 above)
Comment (almeidx) [reply packages/discord.js/src/managers/GuildChannelManager.js:202]: It is not

Comment (vitorlops) [reply packages/discord.js/src/managers/GuildChannelManager.js]: Now it is, thanks for pointing this out! :)

**Comment by vitorlops** [comment]
I agree, I was able to create a webhook with one of the substrings that they said I couldn't. So I [commented](https://github.com/discord/discord-api-docs/pull/4625#issuecomment-1084713065) on the upstream PR, let's see if they reply. 👀
The only one that is not working is "clyde", and it's case-insensitive. So I probably need to also make `name` be `name.toLowerCase()` at the `if`.

**Comment by Markus-Rost** [comment]
> The following restrictions are additionally enforced for usernames:
> 
> 1. Usernames cannot contain the following substrings: '@', '#', ':', '```', 'discord'
> 2. Usernames cannot be: 'everyone', 'here'

The validation you added is only enforced on usernames, but webhooks follow only the nickname restriction so besides clyde they are all valid webhook names.

**Comment by almostSouji** [comment]
The error Discord return
```
DiscordAPIError[50035]: Invalid Form Body
name[WEBHOOK_NAME_INVALID_CONTAINS]: Name cannot contain "clyde"
```
is more verbose than the replacement error introduced here and this isn't as common or restrictive enough of an error to warrant library level validation.

**Comment by vitorlops** [comment]
I see, you are right. I am going to close it then. Thanks for the explanation!

**Pull request closed without merging**